src/webapp/src/accounts/services/users.service.ts

```diff
@@ -85,11 +86,20 @@ export class UsersService extends BaseService<UserEntity> {
    * @param newUser data of the user
    * @param resetPassword flag indicating if password must be resetted during creation. Default to false (see below)
    */
-  async addUser (newUser: NewUserInput, resetPassword = false): Promise<UserEntity | null> {
+  async addUser (newUser: any, resetPassword = false): Promise<UserEntity | null> {
     const user = new UserEntity()
     user.email = newUser.email
+    user.username = newUser.data.username != null && newUser.data.username !== '' ? newUser.data.username : undefined
     user.password = await bcrypt.hash(newUser.password, 12)
-    user.data.name = newUser?.data?.name ?? ''
+    const userSettings = await this.settingsService.getSettings('user') as unknown as SettingsUserJson
```

Comment:
to cast in multiple places, make a settingsService.getUserSettings

Reply:
Done